packages/react-components/react-message-bar-preview/src/components/MessageBarGroup/MessageBarTransition.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { Transition, TransitionStatus } from 'react-transition-group';
+import { MessageBarTransitionContextProvider } from '../../contexts/messageBarTransitionContext';
+import { MessageBarGroupProps } from './MessageBarGroup.types';
+
+const getClassName = (
+  status: TransitionStatus,
+  enterClassName: string,
+  exitClassName: string,
+  animate: MessageBarGroupProps['animate'],
+) => {
+  switch (status) {
+    case 'entering':
+    case 'entered':
+      return animate === 'both' ? enterClassName : '';
+    case 'exiting':
+    case 'exited':
+      return exitClassName;
+  }
+};
+
+/**
+ * Internal component that controls the animation transition for MessageBar components
+ * @internal
+ */
+export const MessageBarTransition: React.FC<{
+  children: React.ReactNode;
+  enterClassName: string;
+  exitClassName: string;
+  animate: MessageBarGroupProps['animate'];
+}> = ({ children, enterClassName, exitClassName, animate, ...rest }) => {
+  return (
+    <Transition timeout={250} {...rest}>
```

Comment:
Please check that we don't fall into `findDOMNode()` scenario, https://github.com/reactjs/react-transition-group/blob/2989b5b87b4b4d1001f21c8efa503049ffb4fe8d/src/Transition.js#L404-L412

Reply:
Yeah good call out, I've made a change in `MessageBarTransition` to clone children to add a ref. I don't think this that bad since that's how RTG expects children too, let me know what you think